internal/mesh/internal/controllers/sidecarproxy/builder/builder_test.go

```diff
@@ -39,3 +42,34 @@ func protoToJSON(t *testing.T, pb proto.Message) string {
 
 	return string(gotJSON)
 }
+
+func JSONToProxyTemplate(t *testing.T, json []byte) *pbmesh.ProxyStateTemplate {
+	t.Helper()
+	proxyTemplate := &pbmesh.ProxyStateTemplate{}
+	m := protojson.UnmarshalOptions{}
+	err := m.Unmarshal(json, proxyTemplate)
+	require.NoError(t, err)
+	return proxyTemplate
+}
+
+func goldenValue(t *testing.T, goldenFile string, actual string, update bool) string {
+	t.Helper()
+	return string(goldenValueBytes(t, goldenFile, actual, update))
+}
+
+func goldenValueBytes(t *testing.T, goldenFile string, actual string, update bool) []byte {
+	t.Helper()
+	goldenPath := filepath.Join("testdata", goldenFile) + ".golden"
+
+	if update {
+		bytes := []byte(actual)
+		err := os.WriteFile(goldenPath, bytes, 0644)
+		require.NoError(t, err)
+
+		return bytes
+	}
+
+	content, err := os.ReadFile(goldenPath)
+	require.NoError(t, err)
+	return content
+}
```

Comment:
I've changed this in my PR to re-use the golden package instead so we probably shouldn't re-add it

Reply:
refactored in dcf54e6373ab4fdb4a32adcfd7fa9b1a70a138a0